packages/contracts/contracts/voting/addresslist/Addresslist.sol

```diff
@@ -46,27 +56,45 @@ abstract contract Addresslist is TimeHelpers {
     }
 
     /// @notice Returns the current length of the address list.
-    /// @return The address list length at the specified block number.
+    /// @return The current address list length.
     function addresslistLength() public view returns (uint256) {
         return _addresslistLengthCheckpoints.latest();
     }
 
-    /// @notice Updates the address list by adding or removing members.
-    /// @param _members The member addresses to be updated.
-    /// @param _enabled Whether to add or remove members from the address list.
-    function _updateAddresslist(address[] calldata _members, bool _enabled) internal {
-        _addresslistLengthCheckpoints.push(
-            _enabled ? _uncheckedAdd : _uncheckedSub,
-            _members.length
-        );
+    /// @notice Internal function to add new addresses to the address list.
+    /// @param _newAddresses The new addresses to be added.
+    function _addAddresses(address[] calldata _newAddresses) internal {
+        for (uint256 i = 0; i < _newAddresses.length; ) {
+            if (isListed(_newAddresses[i])) {
```

Comment:
Unfortunately, this check is not enough. If the `addressA` is not in the checkpoints yet, but `_newAddresses` contains it two times, `isListed` would return false for both of the iterations and you will still increase the length 2 times.

After some thoughts, here is my version how I would do it: Please, include your explanations as the comments. Not sure about the other solution with lower costs. If anybody can think of, let me know. Also, I prefer to not revert if address is already in the checkpoints. Much better in my opinion. Reverting makes no sense and bigger problem. Thoughts? @brickpop @Rekard0 @mathewmeconry 

```
function _addAddresses(address[] calldata _newAddresses) internal {
        uint count = 0;
        for (uint256 i; i < _newAddresses.length; ) {
            (uint oldValue, uint newValue) = _addresslistCheckpoints[_newAddresses[i]].push(1);
            // This means it was actually changed from 0 to 1
            // Even if same address is included 3 times, the below if only be executed 1 time.
            if(oldValue != newValue) {
                count++;
            }
            unchecked {
                ++i;
            }
        }
        _addresslistLengthCheckpoints.push(_uncheckedAdd, count);
    }
```

Reply:
Possibly, test would be needed to confirm my diagnosis and not only for confirming, but for future safety.

Reply:
No, this is not an issue. However, a very good test case.
Added in 1c9b8d1b280af01ad275f51fdb24b96016625644

Reply:
My suggestion should consume less gas(probably 300gass less for 3 loop iteration, but we could neglect this)

Reply:
As mentioned here https://github.com/aragon/core/pull/167/files?diff=split&w=0#r1055229113, I would go with a fail-as-early-as-possible approach. Accordingly, it should revert if an address is passed twice.

E.g., what if the SDK has a bug and includes addresses 10 times, and we don't notice because it accepts it.

Reply:
Resolved after call with @novaknole